windrose/windrose.py

```diff
@@ -0,0 +1,133 @@
+        """
+
+        count = self.calculate_wind_frequency(ndirections,
+                                              vwinds, speed, dirc, direction)
+        count = np.hstack((count[:, [0]], np.diff(count, axis=1)))
+        column_names = (
+                [f'Wind Speed Interval:({vwinds[0]}, {vwinds[1]})'] +
+                [f'Wind Speed Interval:[{vwinds[i]}, '
+                 f'{vwinds[i + 1]})' for i in range(1, len(vwinds) - 1)] +
+                [f'Wind Speed Interval:[{vwinds[-1]}, Inf)']
+        )
+        count_df = pd.DataFrame(count, columns=column_names)
+
+        # center_angle is the angles in which direction bins are centered.
+        # We do not want the 360 to appear, because 0 is already appearing.
+        center_angle = np.linspace(0, 360, ndirections + 1)[:-1]
+        directions = np.mod(
+            self.north - center_angle / 90 * (self.north - self.east), 360)
+        n = 180 / ndirections
+
+        wdirs = [(f"[{np.mod((i - n), 360): .2f}, "
+                  f"{(i + n): .2f})") for i in directions]
+
+        wdirs_series = pd.Series(wdirs)  # Convert wdirs to pandas Series
+
+        count_df.insert(0, 'Direction Interval (Â°)', wdirs_series)
+        #
+        # count_df.insert(0, 'Direction Interval (Â°)', wdirs)
+        count_df.insert(1, 'Avg. Direction', directions)
+        count_df = count_df.sort_values(by='Avg. Direction')
+
+        windzero_frequency = 100 - np.sum(count)
+        windzero_frequency = windzero_frequency * (
+                windzero_frequency / 100 > np.finfo(float).eps)
+        count_df.loc['No Direction'] = (['[0 , 360)', 'Wind Speed = 0'] +
+                                        [''] * (count_df.shape[1] - 3)
+                                        + [str(windzero_frequency)])
+        count_df['total'] = count_df.iloc[:, 2:].sum(axis=1)
+        count_df.loc['Total'] = count_df.iloc[:16, 2:].sum(axis=0)
+
+        return count_df
+
+    def process_season_data(self, ndirections, vwinds, season_df, filename):
+        """
+        Process seasonal data and generate a frequency table.
+
+        Args:
+            ndirections (int): Number of direction bins.
+            vwinds (array-like): Wind speed bins.
+            season_df (pandas.DataFrame): Seasonal data.
+            filename (str): Output filename.
+
+        Returns:
+            None
+        """
+
+        speed = season_df['speed'].to_numpy().reshape(-1, 1)
+        direction = season_df['direction'].to_numpy().reshape(-1, 1)
+
+        dirc = np.mod((self.north - direction) / (
+                self.north - self.east) * 90, 360)
+        dirc = dirc[speed.flatten() > 0]
+        speed = speed[speed.flatten() > 0]
+
+        if len(vwinds) > 0 and vwinds[0] != 0:
+            vwinds = np.insert(vwinds, 0, 0)
+
+        count_df = self.generate_frequency_table(
+            ndirections, vwinds, speed, dirc, direction)
+
+        if 99.9999 <= count_df.iloc[-1, -1] < 100.0001:
```

Comment:
What is the reason for the range?

Reply:
This is written for checking if the total comes as 100%.